src/devices/src/virtio/block/device.rs

```diff
@@ -342,14 +356,65 @@ impl VirtioDevice for Block {
     }
 }
 
+#[derive(Versionize)]
+pub struct BlockState {
+    disk_path: PathBuf,
+    virtio_state: VirtioDeviceState,
+    rate_limiter_state: RateLimiterState,
+}
+
+pub struct BlockConstructorArgs {
+    mem: GuestMemoryMmap,
+}
+
+impl Persist for Block {
+    type State = BlockState;
+    type ConstructorArgs = BlockConstructorArgs;
+    type Error = io::Error;
+
+    fn save(&self) -> Self::State {
+        BlockState {
+            disk_path: self.disk_image_path.clone(),
+            virtio_state: self.state(),
+            rate_limiter_state: self.rate_limiter.save(),
```

Comment:
nit: we should be consistent in naming across components: `dev.state()` vs `rate_limiter.save()`

Reply:
Done. But this led to some interesting casts (e.g. `<Block as VirtioDeviceSave>::save(&self)`) to be able to differentiate between the `save` from `VirtioDeviceSave` and the one from `Persist`.

Reply:
The casts go away if we do it like suggested in https://github.com/firecracker-microvm/firecracker/pull/1755#discussion_r406124926